airflow/utils/db.py

```diff
@@ -1101,7 +1102,7 @@ def check_conn_type_null(session: Session) -> Iterable[str]:
 
     n_nulls = []
     try:
-        n_nulls = session.query(Connection.conn_id).filter(Connection.conn_type.is_(None)).all()
+        n_nulls = session.execute(select(Connection.conn_id).filter(Connection.conn_type.is_(None))).all()
```

Comment:
```suggestion
        n_nulls = session.scalar(select(Connection.conn_id).filter(Connection.conn_type.is_(None)))
```
I feel like `scalar` would be better here

Reply:
And should also be rewritten to not actually select the values in the refacotring PR anyway

Reply:
I think I understand what you mean earlier about not selecting. Looks like it's not much of a change. Something like the below should work:
```python
session.scalars(select(func.count(Connection.id)).where(Connection.conn_type.is_(None))).one()
```
or :
```python
session.scalar(select(func.count(Connection.id)).where(Connection.conn_type.is_(None)))
```
or 
```python
session.scalars(select(func.count()).select_from(Connection).where(Connection.conn_type.is_(None))).one()
```

Reply:
changed it to `session.scalars(select(Connection.conn_id).where(Connection.conn_type.is_(None))).all()`, using `session.scalar` isnt working

Reply:
I guess it didn't work due to the log below that checks len(list)